features/notifications/utils/background/notificationContent.ts

```diff
@@ -1,68 +1,76 @@
-import {
-  ConverseXmtpClientType,
-  DecodedMessageWithCodecsType,
-  GroupWithCodecsType,
-} from "@/utils/xmtpRN/client.types";
+import { DecodedMessageWithCodecsType } from "@/utils/xmtpRN/client.types";
 import {
   getMessageContentType,
   isContentType,
 } from "@/utils/xmtpRN/content-types/content-types";
-import type {
-  MessageId,
-  ReactionContent,
-  ReplyContent,
-} from "@xmtp/react-native-sdk";
+import type { ReactionContent } from "@xmtp/react-native-sdk";
+
+type GetNotificationContentArgs = {
+  message: DecodedMessageWithCodecsType;
+  account: string;
+};
 
 export const getNotificationContent = async (
-  group: GroupWithCodecsType,
-  message: DecodedMessageWithCodecsType
+  args: GetNotificationContentArgs
 ) => {
+  const { message, account } = args;
+
   const supportedContentType = !!getMessageContentType(message.contentTypeId);
   if (!supportedContentType) return;
-  let contentType = message.contentTypeId;
+
+  const contentType = message.contentTypeId;
   const messageContent = message.content();
-  let referencedMessageId: string | undefined;
+
   if (isContentType({ type: "reply", contentType })) {
-    const replyContent = messageContent as ReplyContent;
-    // @todo => implement replies when https://github.com/xmtp/xmtp-node-go/issues/409
-    // is done
-    // referencedMessageId = replyContent.reference;
+    // @todo => implement replies when https://github.com/xmtp/xmtp-node-go/issues/409 is done
     return "REPLY";
   }
 
   if (isContentType({ type: "text", contentType })) {
     return messageContent as string;
-  } else if (isContentType({ type: "remoteAttachment", contentType })) {
+  }
+
+  if (isContentType({ type: "remoteAttachment", contentType })) {
     return "📎 Media";
-  } else if (isContentType({ type: "transactionReference", contentType })) {
+  }
+
+  if (isContentType({ type: "transactionReference", contentType })) {
     return "💸 Transaction";
-  } else if (isContentType({ type: "reaction", contentType })) {
-    let { action, reference, schema, content } =
-      messageContent as ReactionContent;
-    referencedMessageId = reference;
+  }
+
+  // For reactions, we need to check if the message is from the current account
+  if (isContentType({ type: "reaction", contentType })) {
+    const {
+      action,
+      // reference: messageIdAssociatedWithReaction,
+      schema,
+      content,
+    } = messageContent as ReactionContent;
 
     if (action === "added" && schema === "unicode") {
-      // Checking if the group message is from me
-      const isFromMe = await isGroupMessageFromMe(
-        group.client,
-        referencedMessageId
-      );
-      if (!isFromMe) return;
+      // If the message is from the current account, we don't need to show the notification
+      // TODO: This check was there before but not sure if it's needed because we still want to show the notification
+      //  if someone else reacted to a message from the current account?
```

Comment:
great callout i just wanted to prevent 

i send message
i see notification for my message

in my understanding, a reaction would technically be a message from another user - please let me knmow where i'm wrong